Domain/Command{T}.cs

```diff
@@ -175,10 +175,21 @@ private bool CommandHandlerIsRegistered()
             }) != null;
         }
 
-        protected virtual void HandleCommandValidationFailure(TAggregate aggregate, ValidationReport validationReport)
+        protected virtual void HandleCommandValidationFailure(TTarget target, ValidationReport validationReport)
         {
-            ((dynamic) aggregate).HandleCommandValidationFailure((dynamic) this,
-                                                      validationReport);
+            if (target is IEventSourced)
+            {
+                ((dynamic) target).HandleCommandValidationFailure((dynamic) this,
+                                                                  validationReport);
+            }
+            else
```

Comment:
in the case where the target is not event sourced there will not be a HandleCommandValidationFailure. 

currently we elected to use the default that is in place for eventsourced targets. perhaps the requirement for this responsibility should be on an interface a target would implement in the future?